scripts/gen_dockerfile.py

```diff
@@ -0,0 +1,250 @@
+
+    Returns:
+        AppConfig: valid configuration
+    """
+    # Examine app.yaml
+    if not isinstance(raw_config, dict):
+        raise ValueError(
+            'Expected {} contents to be of type "dict", but found type "{}"'.
+            format(args.config, type(raw_config)))
+
+    entrypoint = validation_utils.get_field_value(raw_config, 'entrypoint', str)
+    if not PRINTABLE_REGEX.match(entrypoint):
+        raise ValueError('Invalid character in "entrypoint" field of app.yaml')
+    raw_runtime_config = validation_utils.get_field_value(raw_config, 'runtime_config', dict)
+    python_version = validation_utils.get_field_value(raw_runtime_config, 'python_version', str)
+    dockerfile_python_version = PYTHON_INTERPRETER_VERSION_MAP.get(
+        python_version)
+    if dockerfile_python_version is None:
+        valid_versions = str(sorted(PYTHON_INTERPRETER_VERSION_MAP.keys()))
+        msg = ('Invalid "python_version" field in "runtime_config" section '
+               'of app.yaml.  Valid options are:\n{}').format(valid_versions)
+        raise ValueError(msg)
+
+    # Examine user's files
+    has_requirements_txt = os.path.isfile(
+        os.path.join(args.source_dir, 'requirements.txt'))
+
+    # Compute base image name and Dockerfile contents
+    base_image = args.base_image
+
+    return AppConfig(
+        base_image=base_image,
+        dockerfile_python_version=dockerfile_python_version,
+        entrypoint=entrypoint,
+        has_requirements_txt=has_requirements_txt)
+
+
+def generate_files(app_config):
+    """Generate a Dockerfile and helper files for an application.
+
+    Args:
+        app_config (AppConfig): Validated configuration
+
+    Returns:
+        dict: Map of filename to desired file contents
+    """
+    if app_config.has_requirements_txt:
+        optional_requirements_txt = DOCKERFILE_REQUIREMENTS_TXT
+    else:
+        optional_requirements_txt = ''
+
+    if app_config.entrypoint:
+        # Mangle entrypoint in the same way as the Cloud SDK
+        # (googlecloudsdk/third_party/appengine/api/validation.py)
+        #
+        # We could handle both string ("shell form") and list ("exec
+        # form") but it appears that gcloud only handles string form.
+        entrypoint = app_config.entrypoint
+        if entrypoint and not entrypoint.startswith('exec '):
+            entrypoint = 'exec ' + entrypoint
+        optional_entrypoint = DOCKERFILE_ENTRYPOINT_TEMPLATE.format(
+            entrypoint=entrypoint)
+    else:
+        optional_entrypoint = ''
+
+    dockerfile = DOCKERFILE_TEMPLATE.format(
+        base_image=app_config.base_image,
+        dockerfile_python_version=app_config.dockerfile_python_version,
+        optional_requirements_txt=optional_requirements_txt,
+        optional_entrypoint=optional_entrypoint)
+
+    return {
+        'Dockerfile': dockerfile,
+        '.dockerignore': DOCKERIGNORE,
+    }
+
+
+def gen_dockerfile(args):
```

Comment:
The name of the function doesn't sit well with me, how about `generate_dockerfile_command`?